client/jetpack-cloud/sections/partner-portal/issue-license-v2/licenses-form/index.tsx

```diff
@@ -0,0 +1,295 @@
+	const defaultProductSlugs = useMemo(
+		() => getQueryArg( window.location.href, 'products' )?.toString().split( ',' ),
+		[]
+	);
+
+	useEffect( () => {
+		// Select the slugs included in the URL
+		defaultProductSlugs &&
+			dispatch(
+				addSelectedProductSlugs(
+					// Filter the bundles and select only individual products
+					defaultProductSlugs.filter( ( slug ) => ! isJetpackBundle( slug ) )
+				)
+			);
+
+		// Clear all selected slugs when navigating away from the page to avoid persisting the data.
+		return () => {
+			dispatch( clearSelectedProductSlugs() );
+		};
+	}, [ dispatch, defaultProductSlugs ] );
+
+	const disabledProductSlugs = useSelector< PartnerPortalStore, string[] >( ( state ) =>
+		getDisabledProductSlugs( state, allProducts ?? [] )
+	);
+
+	// We need the suggested products (i.e., the products chosen from the dashboard) to properly
+	// track if the user purchases a different set of products.
+	const suggestedProductSlugs = getQueryArg( window.location.href, 'product_slug' )
+		?.toString()
+		.split( ',' );
+
+	const handleSelectBundleLicense = useCallback(
+		( product: APIProductFamilyProduct ) => {
+			const productBundle = {
+				...product,
+				quantity,
+			};
+			const index = selectedLicenses.findIndex(
+				( item ) => item.quantity === productBundle.quantity && item.slug === productBundle.slug
+			);
+			if ( index === -1 ) {
+				// Item doesn't exist, add it
+				setSelectedLicenses( [ ...selectedLicenses, productBundle ] );
+			} else {
+				// Item exists, remove it
+				setSelectedLicenses( selectedLicenses.filter( ( _, i ) => i !== index ) );
+			}
+		},
+		[ quantity, selectedLicenses, setSelectedLicenses ]
+	);
+
+	const handleShowLicenseOverview = useCallback( () => {
+		// Handle showing the license overview modal here
+	}, [] );
+
+	const onSelectProduct = useCallback(
+		( product: APIProductFamilyProduct ) => {
+			handleSelectBundleLicense( product );
+		},
+		[ handleSelectBundleLicense ]
+	);
+
+	const { isReady } = useSubmitForm( selectedSite, suggestedProductSlugs );
+
+	const onSelectBundle = useCallback(
+		( product: APIProductFamilyProduct ) => {
+			handleSelectBundleLicense( product );
+		},
+		[ handleSelectBundleLicense ]
+	);
+
+	const onClickIssueLicenses = useCallback( () => {
+		handleShowLicenseOverview();
+	}, [ handleShowLicenseOverview ] );
+
+	if ( isLoadingProducts ) {
+		return (
+			<div className="issue-multiple-licenses-form">
+				<div className="issue-multiple-licenses-form__placeholder" />
+			</div>
+		);
+	}
+
+	const selectedSiteDomain = selectedSite?.domain;
+
+	const selectedLicenseCount = selectedLicenses
+		.map( ( license ) => license.quantity )
+		.reduce( ( a, b ) => a + b, 0 );
+
+	const isSelected = ( slug: string ) =>
+		selectedLicenses.findIndex(
+			( license ) => license.slug === slug && license.quantity === quantity
+		) !== -1;
+
+	return (
+		<div className="issue-multiple-licenses-form">
```

Comment:
We should probably change the class name to match the new component name. wdyt?

Reply:
Agree